packages/components/atoms/f-button/src/components/Button.vue

```diff
@@ -14,19 +14,37 @@
         v-bind="$attrs"
         :aria-live="getAriaLive"
         :aria-busy="isLoading"
-        v-on="!isLoading && $listeners"
-    >
+        v-on="!isLoading && $listeners">
         <span
             v-if="isLoading"
             :class="$style['c-spinner']"
             :data-test-id="`${componentType}-spinner`" />
-
         <span
-            :class="[
-                $style['o-btn-text'],
-                (isLoading ? $style['o-btn-text--hidden'] : '')
-            ]">
-            <slot />
+            :class="$style['o-button-content']">
+
+            <span
+                :class="[
+                    (hasLeadingIcon ? $style['o-btn-icon'] : ''),
```

Comment:
Would this work?

```suggestion
                    (hasLeadingIcon ? $style['o-btn-icon', 'o-btn-icon--leading'] : ''),
```

or if not, perhaps:

```suggestion
                    (hasLeadingIcon ? [$style['o-btn-icon'], $style['o-btn-icon--leading']] : ''),
```